### PR TITLE
Use renovate to auto bump deps

### PR DESCRIPTION
**Pull request by jinlee93**
### Summary:
[EDS-758]
- Uses Renovate to auto bump EDS dependencies
- Runs Every Monday 0600 PST
- Supposed to:
 - based off `next` branch
 - assign `edu-design-system` [team](https://docs.renovatebot.com/configuration-options/#reviewers) as reviewers
 - have one concurrent pr 
 - bump non major versions, except for < 0.1.0 and > 0.1.0 and < 1.0.0
 - and some other similar behavior as the [LP renovate](https://github.com/FB-PLP/traject/blob/master/.github/renovate.json5)
### Test Plan:
- tested somewhat working in `jlee/renovate-run` where renovate did create pr's #1403 and #1404 
  - need someone to confirm this is correct behavior since it was 2 concurrent prs and attempted to bump major versions
  - didn't assign `edu-design-system` team 
  - these might be some funky behavior due to not having a ["special token"](https://github.com/renovatebot/github-action#token) so I want to land this to see if it is working correctly with the cron job
  - is correctly based off `next`
- CI runs with renovate prs

[EDS-758]: https://czi-tech.atlassian.net/browse/EDS-758?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jinlee93** [comment]
@jeremiah-clothier Does it need the `enabled: true` config or will it run since the default is not `enabled: false` here? In general do you expect it to work as in the pr summary?

File: .github/renovate.json5 [hunk 1]
```diff
@@ -0,0 +1,85 @@
+{
+  $schema: 'https://docs.renovatebot.com/renovate-schema.json',
+
+  baseBranches: ['next'],
+  enabledManagers: ['npm'],
+  extends: ['config:base', 'group:allNonMajor'],
+  fetchReleaseNotes: false, // Only fetch release notes for breaking changes
+  internalChecksFilter: 'strict', // PRs will be skipped unless a non-pending version is available (applies to "stabilityDays" check only)
```
Comment (jinlee93) [inline .github/renovate.json5:8]: https://github.com/renovatebot/github-action#configurationfile

File: .github/renovate.json5 [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    {
+      // Hold back from creating branches until 3 or more days have elapsed since the version
+      // was released. Note: this only works for packages that provide a release timestamp header
+      // and does not ensure all updates are "stable".
+      matchDatasources: ['npm'],
+      stabilityDays: 3,
+    },
+    {
+      enabled: false,
+    },
```
Comment (jeremiah-clothier) [inline .github/renovate.json5:35]: What does this do?

**Review by jeremiah-clothier — APPROVED** [review]

File: .github/renovate.json5 (hunk 2 above)
Comment (jinlee93) [reply .github/renovate.json5:35]: Cause a bug probably, will remove

File: .github/renovate.json5 [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  baseBranches: ['next'],
+  enabledManagers: ['npm'],
+  extends: ['config:base', 'group:allNonMajor'],
+  fetchReleaseNotes: false, // Only fetch release notes for breaking changes
+  internalChecksFilter: 'strict', // PRs will be skipped unless a non-pending version is available (applies to "stabilityDays" check only)
+  labels: ['npm-dependencies'],
+  prConcurrentLimit: 1,
+  repositories: ['chanzuckerberg/edu-design-system'],
+  rangeStrategy: 'bump', // Always bump minor/patch to latest
+  reviewers: ['team:edu-design-system'],
```
Comment (jeremiah-clothier) [inline .github/renovate.json5:13]: If this syntax doesn't work to assign a group, then one option could be to make edu-design-system codeowners of the package.json file and then they'll get auto assigned.

Comment (jinlee93) [reply .github/renovate.json5:13]: Oooh thats a nice workaround, will keep in mind, thanks

File: .github/renovate.json5 [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    // typescript errors that need to be manually resolved.
+    {
+      matchPackageNames: ['typescript'],
+      groupName: 'typescript',
+    },
+    {
+      matchDepTypes: ['devDependencies'],
+      groupName: 'devDependencies',
+    },
+  ],
```
Comment (jeremiah-clothier) [inline .github/renovate.json5:77]: I wonder if this is overriding the `group:allNonMajor` plugin and causing major updates to be grouped for devDependencies like in https://github.com/chanzuckerberg/edu-design-system/pull/1404

Comment (jinlee93) [reply .github/renovate.json5:77]: Oh makes sense. How would you recommend then grouping `devDeps` separate from prod deps? Or does renovate do that automatically?

Comment (jinlee93) [reply .github/renovate.json5:77]: Should I just land and see what happens on Monday?

Comment (jeremiah-clothier) [reply .github/renovate.json5:77]: I'm not sure off the top of my head.

* What's the end state you're looking for?
* Generally, nonMajor updates are safe, what's the motivation for wanting to group devDeps and prodDeps separately and override the behavior of `group:allNonMajor`?

Comment (jinlee93) [reply .github/renovate.json5:77]: That is a good question. The motivation was that through separation, if a bug rose from an update, the scope of reversion would be smaller. However I do agree that nonMajor updates are generally safe and this worry is probably trivial.

**Pull request merged**